charts/generic-service/templates/controller.yaml

```diff
@@ -22,6 +22,13 @@ spec:
   replicas: {{ .Values.replicas | int }}
   {{- end }}
 
+  {{- if (ne .Values.rollout.revisionHistoryLimit nil) }}
```

Comment:
This can be slightly simplified to:

```suggestion
  {{- if .Values.rollout.revisionHistoryLimit }}
```

Reply:
Correct me if I'm wrong, but I think the version you suggest will result in no `revisionHistoryLimit` being set if `.Values.rollout.revisionHistoryLimit` is `0`, since `0` is falsy.